### ISSUE TITLE
False positive: redundant parens involving partially applied bin op in postfix form

### ISSUE THREAD

**Issue by marat-rkh** [JetBrains/intellij-arend#348]
Redundant parentheses inspection incorrectly reports problem in the following piece of code

```tex
\open Nat

\func foo : (`+ 2) 1 = 3 => idp
```

The code becomes invalid if the user invokes "Unwrap call" fix.

**Comment by marat-rkh** [comment]
Extracted from #342.

**Comment by marat-rkh** [comment]
Fixed in 2dcade7.

**Comment by sxhya** [comment]
Here is another example of false positive redundant parens involving partially applied bin op (this time in prefix form)
```
\func \infixl 6 ++ (a b : Nat -> Nat) => a 1 Nat.+ b 1

\func \infixr 7 ** (a : Nat) (b : Nat) => a Nat.+ b

\func foo => ** 1 ++ (** 3)
```

**Comment by marat-rkh** [comment]
Fixed in 2e9c1f363443584af0e03a6576545f8d10261e2e.
